packages/storage/src/providers/s3/apis/uploadData/multipart/uploadHandlers.ts

```
@@ -171,6 +179,7 @@ export const getMultipartUploadHandlers = (
 					onPartUploadCompletion,
 					onProgress: concurrentUploadsProgressTracker.getOnProgressListener(),
 					isObjectLockEnabled: resolvedS3Options.isObjectLockEnabled,
+					useCRC32Checksum: Boolean(inProgressUpload.finalCrc32),
```

Some thoughts for future improvement. Right now we have no idea if CRC32 checksum fails silently. We need to be deterministic. For example, if we detect the runtime is web but `useCRC32Checksum` is false, we need to throw.

cc @eppjame